framework/src/transfers/MultiAppTransfer.C

```diff
@@ -63,6 +63,8 @@ MultiAppTransfer::MultiAppTransfer(const InputParameters & parameters) :
     _displaced_source_mesh(false),
     _displaced_target_mesh(false)
 {
+  if (Transfer::execFlags() != _multi_app->execFlags())
+      mooseDoOnce(mooseWarning("MultiAppTransfer execute_on flags do not match associated Multiapp execute_on flags"));
```

Comment:
After looking at this we don't need another test, but this doesn't need to be calling the base class version of execFlags